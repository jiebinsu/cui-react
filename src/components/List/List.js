import React from "react";
import PropTypes from "prop-types";
import "./list.scss";
import uuid from "uuid";

const List = ({ type, items }) => {
  const typeMap = {
    bullet: "ul",
    numbered: "ol",
  };
  const Tag = typeMap[type];

  return (
    <Tag>
      {items.map(({ content, ...props }, index) => (
        <li key={uuid()} {...props}>
          {content}
        </li>
      ))}
    </Tag>
  );
};

List.defaultProps = {
  type: "bullet",
  items: [],
};

List.propTypes = {
  type: PropTypes.oneOf(["bullet", "numbered"]),
  items: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.node,
    })
  ),
};

export default List;
