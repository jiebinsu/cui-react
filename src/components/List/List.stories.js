import React from "react";
import List from "./List";

export default {
  title: "Base/List",
  component: List,
};

export const Basic = (args) => <List {...args} />;
Basic.args = {
  items: [
    {
      content: "one",
    },
    {
      content: "two",
    },
    {
      content: "three",
    },
  ],
};
