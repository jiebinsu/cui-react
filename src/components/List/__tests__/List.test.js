import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import List from "../index";

describe("List", () => {
  it("should render default List", () => {
    const wrapper = mount(<List />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<List />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should work with all variants", () => {
    const items = [
      {
        content: "one",
      },
      {
        content: 2,
      },
      {
        content: <div>three</div>,
      },
    ];

    const Variants = () => (
      <div>
        <List type="bullet" items={items} />
        <List type="numbered" items={items} />
      </div>
    );

    const wrapper = mount(<Variants />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<Variants />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
