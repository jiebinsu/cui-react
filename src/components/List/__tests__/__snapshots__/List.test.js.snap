// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render default List 1`] = `<ul />`;

exports[`List should work with all variants 1`] = `
<div>
  <ul>
    <li>
      one
    </li>
    <li>
      2
    </li>
    <li>
      <div>
        three
      </div>
    </li>
  </ul>
  <ol>
    <li>
      one
    </li>
    <li>
      2
    </li>
    <li>
      <div>
        three
      </div>
    </li>
  </ol>
</div>
`;
